### DISCUSSION TITLE
Feature Request: In the REST API allow to filter models by filename

### DISCUSSION THREAD

**Discussion by QuietNoise** [civitai/civitai#404] [Ideas]
It would help immensely if we could find a civitai model record in API based on the model's file name. 
I.e. in the https://civitai.com/api/v1/models endpoint an optional argument filename=somefile.safetensor would filter results to a model which has this file listed in its model versions.

**Comment by BlafKing** [comment]
An alternative is to generate the SHA256 hash of the file and then send that as an API request to:
`https://civitai.com/api/v1/model-versions/by-hash/hash_here`
